src/controllers/resource-store/index.tsx

```diff
@@ -56,11 +56,14 @@ export const actions: Actions = {
       type,
       key,
     });
+    const data = getNewSliceData(prevSlice.data);
 
     dispatch(
       setResourceState(type, key, {
         ...prevSlice,
-        data: getNewSliceData(prevSlice.data),
+        data,
+        error: null,
+        promise: Promise.resolve(data),
```

Comment:
q/ will this re-trigger suspense in some weird way, needs to be covered in enzyme tests?

Reply:
Current enzyme cannot test suspense. I haven't been able to find a work around for this.

I think we can reason about the use of `promise` as in `throw promise`- Changes in the update() would re-render components that have `useResource()` hook. In this render the `promise` may be different but the `loading` is `false`. Therefore we would not `throw` such a promise any more than we would `throw` the promise that remains after remote async loading once that load is complete.